tableaudocumentapi/datasource.py

```diff
@@ -178,16 +179,17 @@ def fields(self):
         return self._fields
 
     def _get_all_fields(self):
-        column_objects = [_column_object_from_column_xml(self._datasourceTree, xml)
-                          for xml in self._datasourceTree.findall('.//column')]
-        existing_fields = [x.id for x in column_objects]
-        metadata_fields = (x.text
-                           for x in self._datasourceTree.findall(".//metadata-record[@class='column']/local-name"))
-
-        missing_fields = (x for x in metadata_fields if x not in existing_fields)
-        column_objects.extend((
-            _column_object_from_metadata_xml(_get_metadata_xml_for_field(self._datasourceTree, field_name))
-            for field_name in missing_fields
-        ))
-
-        return FieldDictionary({k: v for k, v in column_objects})
+        column_field_objects = self._get_column_objects()
+        existing_column_fields = [x.id for x in column_field_objects]
+        metadata_field_objects = (x for x in self._get_metadata_objects() if x.id not in existing_column_fields)
```

Comment:
metadata_only_field_objects -- or some hint that we have to find these a different way